cartographer_grpc/framework/rpc.h

```diff
@@ -36,21 +40,30 @@ class Rpc {
     Rpc* rpc;
   };
 
-  Rpc(const RpcHandlerInfo& rpc_handler_info);
+  Rpc(int method_index, ::grpc::ServerCompletionQueue* server_completion_queue,
+      const RpcHandlerInfo& rpc_handler_info, Service* service);
 
+  int method_index() const { return method_index_; }
+  ::grpc::ServerCompletionQueue* server_completion_queue();
   ::grpc::internal::RpcMethod::RpcType rpc_type() const;
+  Service* service() { return service_; }
   ::grpc::ServerContext* server_context() { return &server_context_; }
-  ::grpc::internal::ServerAsyncStreamingInterface* responder();
-  RpcState* GetState(State state);
+  ::grpc::internal::ServerAsyncStreamingInterface* streaming_interface();
+  RpcState* GetRpcState(State state);
+  const RpcHandlerInfo& rpc_handler_info() const { return rpc_handler_info_; }
 
   ::google::protobuf::Message* request() { return request_.get(); }
   ::google::protobuf::Message* response() { return response_.get(); }
 
  private:
   Rpc(const Rpc&) = delete;
   Rpc& operator=(const Rpc&) = delete;
+  void InitializeResponders(::grpc::internal::RpcMethod::RpcType rpc_type);
 
+  int method_index_;
+  ::grpc::ServerCompletionQueue* server_completion_queue_;
   RpcHandlerInfo rpc_handler_info_;
+  Service* service_;
```

Comment:
As discussed offline, consider moving this to RpcState.

Reply:
Done.